### PR TITLE
Move tests for app protocols from k/k to Ingress-GCE repo

### PR DESCRIPTION
**Pull request by rramkumar1**
The PR is split into two commits:

1. Extract echoheaders response body into public struct so it can be used elsewhere
2. Porting of tests

Note that the PR to remove the tests from k/k is https://github.com/kubernetes/kubernetes/pull/75143

### CODE REVIEW

File: pkg/e2e/fixtures.go [hunk 1]
```diff
@@ -74,6 +85,20 @@ func CreateEchoService(s *Sandbox, name string, annotations map[string]string) (
 	return pod, service, nil
 }
 
+// UpdateService updates an existing service with new annotations.
+func UpdateService(s *Sandbox, name string, annotations map[string]string) error {
```
Comment (MrHohn) [inline pkg/e2e/fixtures.go]: nit: maybe `UpdateServiceAnnotations`?

File: pkg/fuzz/features/app_protocol.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	var r echo.ResponseBody
+	if err := json.Unmarshal(body, &r); err != nil {
+		return fuzz.CheckResponseContinue, err
+	}
+
+	switch proto {
+	case annotations.ProtocolHTTPS:
+		if !(strings.HasPrefix(r.HTTPVersion, "1.") && r.TLS) {
+			return fuzz.CheckResponseContinue, fmt.Errorf("Expected HTTP/1.x response with TLS configured on request: %+v", r)
```
Comment (MrHohn) [inline pkg/fuzz/features/app_protocol.go]: nit: Expected -> expected

File: cmd/e2e-test/app_protocols_test.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+			t.Parallel()
+
+			ctx := context.Background()
+
+			svcAnnotation := map[string]string{
+				annotations.ServiceApplicationProtocolKey: tc.annotationVal,
+			}
+			_, _, err := e2e.CreateEchoService(s, "service-1", svcAnnotation)
+			if err != nil {
+				t.Fatalf("error creating echo service: %v", err)
```
Comment (MrHohn) [inline cmd/e2e-test/app_protocols_test.go]: nit: error -> Error (and a couple below)

File: cmd/e2e-test/app_protocols_test.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+			t.Logf("Waiting for GCLB resources to be deleted (%s/%s)", s.Namespace, ing.Name)
+			if err := e2e.WaitForGCLBDeletion(ctx, Framework.Cloud, gclb, deleteOptions); err != nil {
+				t.Errorf("e2e.WaitForGCLBDeletion(...) = %v, want nil", err)
+			}
+			t.Logf("GCLB resources deleted (%s/%s)", s.Namespace, ing.Name)
+		})
+	}
+}
+
+func TestAppProtocolTransition(t *testing.T) {
```
Comment (MrHohn) [inline cmd/e2e-test/app_protocols_test.go]: Is the plan to keep this running for now but fail, until we re-vendor?

File: pkg/e2e/fixtures.go (hunk 1 above)
Comment (rramkumar1) [reply pkg/e2e/fixtures.go]: Rebased so now I use EnsureEchoService

File: pkg/fuzz/features/app_protocol.go (hunk 2 above)
Comment (rramkumar1) [reply pkg/fuzz/features/app_protocol.go]: Done

File: cmd/e2e-test/app_protocols_test.go (hunk 3 above)
Comment (rramkumar1) [reply cmd/e2e-test/app_protocols_test.go]: Done

File: cmd/e2e-test/app_protocols_test.go (hunk 4 above)
Comment (rramkumar1) [reply cmd/e2e-test/app_protocols_test.go]: I will omit it from the regex which selects  tests to run until we have the revendor.

**Comment by rramkumar1** [comment]
@bowei This is ready for final review

**Pull request merged**